### PR TITLE
Protocol Script Template

### PR DESCRIPTION
**Pull request by ramifarawi**
## overview

This is a template python script which should be used both externally and internally for any protocol scripts to be added to the Opentrons Protocol Library. The script lays out the conventions and styles to use for protocols uploaded to the library with examples and descriptions. The template also lays out the order in which scripts should be written.

### CODE REVIEW

File: Template/protocol_template.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        "custom_variable1",
+        "custom_variable2")
+
+    if not 1 <= custom_variable1 <= 12:
+        raise Exception("Enter a value between 1-12")
+
+    # define all custom variables above here with descriptions:
+
+    # number of samples
+    custom_variable1 = custom_variable1
```
Comment (chazmanc) [inline Template/protocol_template.py:27]: I might doing something like:
`custom_variable1 = _custom_variable1`

But open to discussion

File: Template/protocol_template.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    - Custom drop tip functions
+    - Custom Tip tracking functions
+    - Custom Trash tracking functions
+    - Slow tip withdrawal
+
+    For any functions in your protocol, describe the function as well as
+    describe the parameters which are to be passed in as a docstring below
+    the function (see below).
+
+    def pick_up(pipette):
```
Comment (chazmanc) [inline Template/protocol_template.py:120]: nice example 😎

**Review by chazmanc — APPROVED** [review]
Looks good to me! Would love to hear @eskila 's thoughts

File: Template/protocol_template.py (hunk 1 above)
Comment (eskila) [reply Template/protocol_template.py]: That's a good idea, that makes it more explicit that we're assigning some input variable to a variable that we will use in our script.

Also the comment could say "e.g. number of samples" instead

File: Template/protocol_template.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    labware_name = ctx.load_labware('{loadname}', '{slot number}')
+
+    If loading labware on a module, load with:
+
+    labware_name = module_name.load_labware('{loadname}')
+    where module_name is defined above.
+
+    '''
+
+    # load tipracks
```
Comment (eskila) [inline Template/protocol_template.py:62]: I noticed that in some protocols the pipettes are loaded first, and then the tipracks are heuristically determined based on what volumes the pipettes can handle. Do we want that in our template?

**Comment by eskila** [comment]
Is there a way to find out why Travis is unhappy?

File: Template/protocol_template.py (hunk 3 above)
Comment (eskila) [reply Template/protocol_template.py:62]: We could load the pipettes first, and then the writer could make that call themselves.

**Review by eskila — APPROVED** [review]

**Pull request merged**